### PR TITLE
List all server locators [10725]

### PR DESCRIPTION
**Pull request by EduPonz**
With this PR, `fastdds discovery` lists all the locators specified for the server.

Signed-off-by: EduPonz <eduardoponz@eprosima.com>

### CODE REVIEW

File: tools/fds/server.cpp [hunk 1]
```diff
@@ -228,10 +227,20 @@ int main (
...
-        cout << "  Participant Type    " << rtps.builtin.discovery_config.discoveryProtocol << endl;
+        cout << "  Server Addresses:   ";
+        for (auto locator_it = att->rtps.builtin.metatrafficUnicastLocatorList.begin();
+                locator_it != att->rtps.builtin.metatrafficUnicastLocatorList.end();)
+        {
+            cout << *locator_it;
+            if (++locator_it != att->rtps.builtin.metatrafficUnicastLocatorList.end())
+            {
+                cout << std::endl << "                      ";
+            }
```
Comment (JLBuenoLopez-eProsima) [inline tools/fds/server.cpp:241]:
```suggestion
        for (auto locator : att->rtps.builtin.metatrafficUnicastLocatorList)
        {
            cout << locator << std::endl << "                      ";
        }
```

I suggest using the ranged for. Also, LocatorList class have operator `<<` overloaded so we can also take advantage of this.

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]

**Comment by JLBuenoLopez-eProsima** [comment]
Please, check also the following Discovery Server v2.0 tests that are failing and it is probably due to the changes implemented in this PR:

* 22: environment variable setup
* 23: fast discovery server tool

File: tools/fds/server.cpp (hunk 1 above)
Comment (EduPonz) [reply tools/fds/server.cpp:241]: I actually was doing that in the beginning and then at some point switched to the iterators to see if that solved another problem I was having. I'll go back to range for 👌🏼

Comment (EduPonz) [reply tools/fds/server.cpp:241]: After internal discussion, we have decided to leave the iterator to be able to do different things for the last element

**Comment by EduPonz** [comment]
This tests should be failing since #1764 . PR eProsima/Discovery-Server#25 was issued to deal with that change

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM

**Comment by EduPonz** [comment]
@richiprosima please test discovery-server

**Pull request merged**